File: 3sim/org.palladiosimulator.blockchainsystems.threesim/src/main/java/org/palladiosimulator/blockchainsystems/doublespending/simulation/termination/LongestChainExceededMaxLengthCondition.java
package org.palladiosimulator.blockchainsystems.doublespending.simulation.termination;

public class LongestChainExceededMaxLengthCondition {

	private long _currentLength = 0;
	private final long _maxAllowedLength;
	
	public LongestChainExceededMaxLengthCondition(long maxAllowedLength) {
		_maxAllowedLength = maxAllowedLength;
	}
	
	public long getCurrentLength() {
		return _currentLength;
	}
	
	public boolean hasLengthExceeded() {
		return _currentLength >= _maxAllowedLength;
	}
	
	public void onBlockAppended(long blockPosition) {
		_currentLength = Math.max(blockPosition, _currentLength);
	}
}

File: 3sim/org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/simulation/termination/LongestChainExceededMaxLengthCondition.kt
package org.palladiosimulator.blockchainsystems.core.simulation.termination

import kotlin.math.max

class LongestChainExceededMaxLengthCondition(
  private val maxAllowedLength: Long
) {
  var currentLength: Long = 0
    private set

  fun hasLengthExceeded(): Boolean {
    return currentLength >= maxAllowedLength
  }

  fun onBlockAppended(blockPosition: Long) {
    currentLength = max(blockPosition, currentLength)
  }
}